### PR TITLE
Refactor: extract verify_leaf_storage_node & verify_leaf_state_node functions

### PR DESCRIPTION
**Pull request by Pasifaee**
Extracted verify_leaf_X_node functions.

### CODE REVIEW

File: lib/src/lib.nr [hunk 1]
```diff
@@ -255,6 +279,32 @@ pub fn verify_node_hash<N>(node: [u8; N], hash: [u8; KEY_LENGTH])
...
+/// Verifies that a middle node hashes to extracted hash. Extracts the next hash and advances the key pointer.
+///
+/// # Arguments
+/// * `extracted_hash` - Hash extracted from the preceding node
+/// * `key_ptr` - Pointer to current nibble in the key
+/// * `key_nibbles` - Array containing the nibbles of the key
+/// * `path` - RLP-encoded proof path
+/// * `depth` - Depth of proof path
+/// * `cur_depth` - Current depth in the proof path
+fn verify_middle_node<PROOF_LEN>(extracted_hash: &mut [u8; KEY_LENGTH], key_ptr: &mut u64, key_nibbles: [u8; NIBBLE_LENGTH], path: [u8; PROOF_LEN], depth: u64, cur_depth: u64) {
```
Comment (ax0) [inline lib/src/lib.nr]: Remove this function.

File: lib/src/lib.nr [hunk 2]
```diff
@@ -85,12 +83,31 @@ impl<PROOF_LEN, MAX_VALUE_LEN> TrieProof<32, PROOF_LEN, MAX_VALUE_LEN>
...
+    /// # Arguments
+    /// * `hash` - Hash extracted from the preceding node
+    /// * `value` - Value to be verified
+    /// * `value_length` - Length of the value
+    /// * `key` - Key to be verified
+    /// * `key_ptr` - Pointer to current nibble in the key
+    /// * `key_nibbles` - Array containing the nibbles of the key
+    /// * `path` - RLP-encoded proof path
+    /// * `depth` - Depth of proof path
+    pub fn verify_leaf_storage_node(self, hash: [u8; KEY_LENGTH], key_len: Field, key_ptr: u64, key_nibbles: [u8; NIBBLE_LENGTH])
```
Comment (ax0) [inline lib/src/lib.nr]: Could you rename this to `verify_storage_leaf_node`?

File: lib/src/lib.nr [hunk 3]
```diff
@@ -145,11 +155,30 @@ impl<PROOF_LEN, MAX_VALUE_LEN> TrieProof<20, PROOF_LEN, MAX_VALUE_LEN>
...
+    /// # Arguments
+    /// * `extracted_hash` - Hash extracted from the preceding node
+    /// * `value` - Value to be verified
+    /// * `value_length` - Length of the value
+    /// * `key` - Key to be verified
+    /// * `key_ptr` - Pointer to current nibble in the key
+    /// * `key_nibbles` - Array containing the nibbles of the key
+    /// * `path` - RLP-encoded proof path
+    /// * `depth` - Depth of proof path
+    fn verify_leaf_state_node(self, extracted_hash: [u8; KEY_LENGTH], key_len: Field, key_ptr: u64, key_nibbles: [u8; NIBBLE_LENGTH])
```
Comment (ax0) [inline lib/src/lib.nr]: As above, maybe rename this to `verify_state_leaf_node`?

**Review by ax0 — CHANGES_REQUESTED** [review]

File: lib/src/lib.nr (hunk 2 above)
Comment (Pasifaee) [reply lib/src/lib.nr]: Sure, renamed

File: lib/src/lib.nr (hunk 3 above)
Comment (Pasifaee) [reply lib/src/lib.nr]: Renamed

File: lib/src/lib.nr (hunk 1 above)
Comment (Pasifaee) [reply lib/src/lib.nr]: Sorry, removed

**Comment by ax0** [comment]
LGTM. Thanks!

**Pull request merged**